src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/DoNotDecreaseInheritedMemberVisibilityTests.cs

```diff
@@ -12,12 +10,295 @@ public class DoNotDecreaseInheritedMemberVisibilityTests : DiagnosticAnalyzerTes
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicDoNotDecreaseInheritedMemberVisibilityAnalyzer();
+            return new DoNotDecreaseInheritedMemberVisibilityAnalyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpDoNotDecreaseInheritedMemberVisibilityAnalyzer();
+            return new DoNotDecreaseInheritedMemberVisibilityAnalyzer();
+        }
+
+        [Fact]
+        public void DecreaseCSharpMemberVisibility()
+        {
+            VerifyCSharp(
+@"public class BaseClass
+{
+    public int MyProp3;
+}
+
+public class DerivedClass : BaseClass
+{
+    private new int MyProp3;
+}", GetCSharpCA2222RuleNameResultAt(8, 21));
+
+            VerifyCSharp(
+@"public class BaseClass
+{
+    public int MyProp3 { get; set; }
+}
+
+public class DerivedClass : BaseClass
+{
+    internal new int MyProp3 { get; set; }
+}", GetCSharpCA2222RuleNameResultAt(8, 22), GetCSharpCA2222RuleNameResultAt(8, 32), GetCSharpCA2222RuleNameResultAt(8, 37));
```

Comment:
I agree that would be a better experience. But for the reasons I mentioned above, we want maximum FxCop compatibility for the first release, even at the cost of a sub-optimal experience: to reduce the friction of migrating, and to facilitate testing. If you can duplicate the FxCop behavior without significant extra effort, please do so.